cli/crates/federated-dev/src/dev/bus.rs

```diff
@@ -35,14 +37,13 @@ pub(crate) type ComposeSender = mpsc::Sender<ComposeMessage>;
 /// Receive channel for the composer to control its state and actions
 pub(crate) type ComposeReceiver = mpsc::Receiver<ComposeMessage>;
 
-/// Send half of channel for the server to send requests to the router actor
-pub(crate) type RequestSender = mpsc::Sender<(engine::Request, RequestHeaders, ResponseSender)>;
-
-/// Receive half of channel for the router actor to receive requests
-pub(crate) type RequestReceiver = mpsc::Receiver<(engine::Request, RequestHeaders, ResponseSender)>;
+/// Send half of the gateway watch channel
+pub(crate) type GatewaySender = watch::Sender<Option<Arc<Gateway>>>;
 
-/// Send half of channel for the router actor to send responses
-pub(crate) type ResponseSender = oneshot::Sender<RouterResult<engine_v2::CacheableResponse>>;
+/// Receive half of the gateway watch channel.
+///
+/// Anything part of the system that needs access to the gateway can use this
+pub(crate) type GatewayReceiver = watch::Receiver<Option<Arc<Gateway>>>;
```

Comment:
maybe `GatewayWatcher`? Receiver sounds a bit weird to me as you do `receiver.borrow().clone()` to "receive" something

Reply:
Yeah, wasn't happy with the name either - good idea